Beehiv fix

stuartbreckenridge opened the pull request:
Fixes #4399 

![Simulator Screenshot - iPhone 16 Pro Max - 2024-12-31 at 17 10 55](https://github.com/user-attachments/assets/9e496677-4aee-4fa9-b4e7-e3140bdeef5d)

stuartbreckenridge committed:
Adds secrets

stuartbreckenridge committed:
🩹 Stop beehiv blog style slipping into summaries

brentsimmons commented on RSCore/Sources/RSCore/Shared/String+RSCore.swift:
```diff
@@ -195,7 +195,15 @@ public extension String {
 	/// - Note: Doesn't work correctly with nested tags of the same name.
 	private func removingTagAndContents(_ tag: String) -> String {
-		return self.replacingOccurrences(of: "<\(tag).+?</\(tag)>", with: "", options: [.regularExpression, .caseInsensitive])
+		let pattern = "<\(tag)>.*?<\\/\(tag)>"
+		if let regex = try? NSRegularExpression(pattern: pattern, options: [.dotMatchesLineSeparators, .caseInsensitive]) {
+			let range = NSRange(location: 0, length: self.utf16.count)
+			let modifiedString = regex.stringByReplacingMatches(in: self, options: [], range: range, withTemplate: "")
+			return modifiedString
+		} else {
```
Would you add a comment here explaining the approach? I’m no regex expert, and it would be helpful to me. (One question I have: wouldn’t there be a way to alter the pattern we were already using to include line separators?)

stuartbreckenridge replied:
Code comments added in d2364ff.

stuartbreckenridge committed:
🔥 Remove SecretKey.swift

Likely need to add this to .gitignore.

stuartbreckenridge commented on Modules/Secrets/Sources/Secrets/SecretKey.swift:
```diff
@@ -0,0 +1,64 @@
+// Generated by SecretKey.swift.gyb
```
Done.

stuartbreckenridge committed:
📝 Adds code comments for the regex used.

brentsimmons commented on RSCore/Sources/RSCore/Shared/String+RSCore.swift:
```diff
@@ -189,13 +189,37 @@ public extension String {
 	private func removingTagAndContents(_ tag: String) -> String {
-		return self.replacingOccurrences(of: "<\(tag).+?</\(tag)>", with: "", options: [.regularExpression, .caseInsensitive])
+		let pattern = "<\(tag)>.*?<\\/\(tag)>"
+		if let regex = try? NSRegularExpression(pattern: pattern, options: [.dotMatchesLineSeparators, .caseInsensitive]) {
+			let range = NSRange(location: 0, length: self.utf16.count)
+			let modifiedString = regex.stringByReplacingMatches(in: self, options: [], range: range, withTemplate: "")
+			return modifiedString
+		} else {
```
Instead of trying one thing and then a second if it fails, could we just modify the original pattern — make it something like this?

```
<\(tag)>[\\s\\S]*?</\(tag)>
```

I’m not sure the above is totally right — but the idea is to look for zero or more of whitespace/non-whitespace between the tags, which should include line separators.

stuartbreckenridge replied:
That approach also works. See 9b508e0.

No easy fix for nested tags though.

stuartbreckenridge committed:
🩹 Change regex pattern to use [\\s\\S]

📝 Code commentary updated to reflect change to `[\\s\\S]`.
✅ Tested on both default and Beehiiv feeds.
